materialize-starburst/starburst.go

```diff
@@ -0,0 +1,430 @@
+	_ sql.Fence,
+	tables []sql.Table,
+	open pm.Request_Open,
+) (_ m.Transactor, err error) {
+	var cfg = ep.Config.(*config)
+	var dialect = starburstDialect(cfg.Schema)
+	var templates = renderTemplates(dialect)
+
+	var transactor = &transactor{
+		cfg: cfg,
+	}
+	if transactor.updateDelay, err = sql.ParseDelay(cfg.Advanced.UpdateDelay); err != nil {
+		return nil, err
+	}
+
+	// Establish connections.
+	transactor.load.conn, err = connectToDb(ctx, cfg.ToURI())
+	if err != nil {
+		return nil, fmt.Errorf("connection for load failed: %w", err)
+	}
+	transactor.store.conn, err = connectToDb(ctx, cfg.ToURI())
+	if err != nil {
+		return nil, fmt.Errorf("connection for store failed: %w", err)
+	}
+
+	for _, table := range tables {
+		materializationSpec := open.Materialization.Bindings[table.Binding]
+		if err = transactor.addBinding(ctx, materializationSpec, table, templates); err != nil {
+			return nil, fmt.Errorf("adding binding %v failed: %w", table, err)
+		}
+	}
+
+	s3Config := s3config{AWSAccessKeyID: cfg.AWSAccessKeyID, AWSSecretAccessKey: cfg.AWSSecretAccessKey, Region: cfg.Region, Bucket: cfg.Bucket}
+	transactor.s3Operator, err = NewS3Operator(s3Config)
+	if err != nil {
+		return nil, fmt.Errorf("creating s3 operator: %w", err)
+	}
+
+	return transactor, nil
+}
+
+type binding struct {
+	target              sql.Table
+	materializationSpec *pf.MaterializationSpec_Binding
+	load                struct {
+		createTempTable string
+		dropTempTable   string
+		loadQuery       string
+	}
+	store struct {
+		createTempTable string
+		dropTempTable   string
+		mergeIntoTarget string
+	}
+}
+
+func (t *transactor) addBinding(ctx context.Context, materializationSpec *pf.MaterializationSpec_Binding, target sql.Table, templates templates) error {
+	var d = new(binding)
+	var err error
+	d.target = target
+
+	if d.load.createTempTable, err = sql.RenderTableTemplate(target, templates.createLoadTempTable); err != nil {
+		return fmt.Errorf("createLoadTempTable template: %w", err)
+	}
+	if d.load.dropTempTable, err = sql.RenderTableTemplate(target, templates.dropLoadTempTable); err != nil {
+		return fmt.Errorf("dropLoadTempTable template: %w", err)
+	}
+	if d.load.loadQuery, err = sql.RenderTableTemplate(target, templates.loadQuery); err != nil {
+		return fmt.Errorf("loadQuery template: %w", err)
+	}
+
+	if d.store.createTempTable, err = sql.RenderTableTemplate(target, templates.createStoreTempTable); err != nil {
+		return fmt.Errorf("createStoreTempTable template: %w", err)
+	}
+	if d.store.dropTempTable, err = sql.RenderTableTemplate(target, templates.dropStoreTempTable); err != nil {
+		return fmt.Errorf("dropStoreTempTable template: %w", err)
+	}
+	if d.store.mergeIntoTarget, err = sql.RenderTableTemplate(target, templates.mergeIntoTarget); err != nil {
+		return fmt.Errorf("mergeIntoTarget template: %w", err)
+	}
+
+	d.materializationSpec = materializationSpec
+	t.bindings = append(t.bindings, d)
+
+	// Drop existing temp tables
```

Comment:
I understand this cleanup is safe because the staged files aren't being touched, and these temporary tables are scoped to exist _just_ while a specific MERGE is happening (create table / load / drop table flow in Acknowledge)

Reply:
Yes. Drop table does not remove files so we can recreate table from the same path.